pennylane/operation.py

```diff
@@ -900,6 +910,48 @@ def matrix(self):
         # over the defined wires.
         return functools.reduce(np.kron, U_list)
 
+    def prune(self):
+        """Returns a pruned tensor product of observables by removing :class:`~.Identity` instances from
+        the observables building up the :class:`~.Tensor`.
+
+        The return_type attribute is preserved while pruning.
+
+        If the tensor product only contains one observable, then this observable instance is
+        returned.
+
+        Note that this way this method might return an instance that is not a :class:`~.Tensor`
+        instance.
+
+        **Example:**
+
+        1.
+
+        >>> O = qml.PauliZ(0) @ qml.Identity(1) @ qml.PauliZ(2)
+        >>> pruned_obs = O.prune()
+        <pennylane.operation.Tensor at 0x7fc1642d1590
```

Comment:
Should this show output? Normally when you do an assignment on the Python prompt, there is no output displayed.

Reply:
This became:
```python
        >>> [(o.name, o.wires) for o in O.prune().obs]
        [('PauliZ', [0]), ('PauliZ', [2])]
```

Reply:
Removed the assignment here, thanks!